drivers/eeprom/eeprom_st25dv.c

```diff
@@ -0,0 +1,970 @@
+/*
+ * Copyright (c) 2023 Kickmaker
+ * Copyright (c) 2021 Martin Schröder <info@swedishembedded.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_st25dv
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/eeprom.h>
+#include <zephyr/drivers/eeprom/eeprom_st25dv.h>
+#include <soc.h>
+
+#define LOG_LEVEL CONFIG_EEPROM_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(eeprom_st25dv);
+
+/** @brief ST25DV registers */
+typedef enum {
```

Comment:
Please avoid using typedefs for stuff like this. I'd prefer either just an enum or even better just `#define`.

Reply:
Updated to use #define